src/fides/api/ops/service/connectors/saas_query_config.py

```diff
@@ -51,42 +51,76 @@ def __init__(
         self.data_protection_request = data_protection_request
         self.privacy_request = privacy_request
         self.action: Optional[str] = None
+        self.current_request: Optional[SaaSRequest] = None
 
-    def get_request_by_action(self, action: str) -> Optional[SaaSRequest]:
+    def get_read_requests_by_identity(self) -> List[SaaSRequest]:
         """
-        Returns the appropriate request config based on the
-        current collection and preferred action (read, update, delete)
+        Returns the appropriate request configs based on the current collection and identity
         """
         try:
-            # store action name for logging purposes
-            self.action = action
             collection_name = self.node.address.collection
-            request = self.endpoints[collection_name].requests[action]  # type: ignore
-            logger.info(
-                "Found matching endpoint to %s '%s' collection", action, collection_name
+            requests = self.endpoints[collection_name].requests
+            read_requests = (
+                requests.read if isinstance(requests.read, list) else [requests.read]
             )
-            return request
+            filtered_requests = self._requests_using_identity(read_requests)
+            # return all the requests if none contained an identity reference
+            return read_requests if not filtered_requests else filtered_requests
         except KeyError:
             logger.info(
-                "The '%s' action is not defined for the '%s' endpoint in %s",
-                action,
+                "The 'read' action is not defined for the '%s' endpoint in %s",
                 collection_name,
                 self.node.node.dataset.connection_key,
             )
-            return None
+            return []
+
+    def _requests_using_identity(
+        self, requests: List[SaaSRequest]
+    ) -> List[SaaSRequest]:
+        """Filters for the requests using the provided identity"""
+
+        return [
+            request
+            for request in requests
+            if any(
+                param_value
+                for param_value in request.param_values or []
+                if param_value.identity == self._get_identity()
+            )
+        ]
+
+    def get_erasure_request_by_action(
+        self, action: Literal["update", "delete"]
+    ) -> Optional[SaaSRequest]:
+        """
+        Returns the appropriate request config based on the
+        current collection and preferred erasure action (update or delete)
+        """
+
+        collection_name = self.node.address.collection
+        request: Optional[SaaSRequest] = getattr(
+            self.endpoints[collection_name].requests, action
+        )
+        if request:
+            logger.info(
```

Comment:
```
if request:
    logger.info(
        "Found matching endpoint to %s '%s' collection", action, collection_name
    )
else:
    logger.error(
        "Unable to find matching endpoint to %s '%s' collection",
        action,
        collection_name,
    )
```

Reply:
I realized this too when I started working on the phone number identity changes, I changed it on my branch but I can change it to a logger here too since this is where the change was introduced.